astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -675,12 +675,14 @@ def test_all(self):
             np.all(self.q)
 
     # NUMPY_LT_1_25
```

Comment:
I think we can remove these comments now, as the test will stay with us until support for numpy<2 is dropped, and we'll never need to remove the warning filters.

Reply:
🤔 

Unless we have no intention to keep testing deprecated features after we drop support for NumPy 1.25. Let's see what @mhvk says.

Reply:
As to "should we keep testing them at all when deprecated?", they will continue be available through `quantity` or `utils.masked` as long as they are available in Numpy, and thus should still be tested, with the warnings.
Actually I'd find it preferable to test that they _do_ emit the `DeprecationWarning` as of 1.25 (i.e. `with pytest.warns` rather than `filterwarnings`).
At least in `masked` – maybe not so much for `units` where only raising an error is checked anyway (I am a bit surprised this still needs the warnings filter then – is the warning raised before the exception)?
Could in principle even be done conditionally now, see https://github.com/astropy/astropy/pull/14510/files#r1131609447

Reply:
I *think* with pytest 8, it catches everything now (but I don't have time to confirm) and the ignore also makes it explicit that we are aware there is a warning and we chose to ignore it, so I don't see any harm in keeping it.

Reply:
> Actually I'd find it preferable to test that they do emit the DeprecationWarning as of 1.25 (i.e. with pytest.warns rather than filterwarnings)

I feel that's out of scope for this PR but I agree that's preferable. I'll plan to do this when we don't have more urgent fixes in the queue (hopefully sometimes next week).

Reply:
Re: `pytest.warns` -- I am not sure if doing this is worth the contract money. I think `filterwarnings` is fine since it means we don't care about the warning because it is from numpy and not something we controls. If numpy suddenly changes that warning message or whatever, we also don't care.

Reply:
Makes sense !

Reply:
Yes, I did not care about what specific warning Numpy issues either, just that the present form is oblivious whether this is marked deprecated at all. Of course it will never happen that Numpy 1.27.0 suddenly undeprecates these functions, but a lot of stuff is testing things that will never happen™.
Not worth investing much time in now anyway.